src/hist/basehist.py

```diff
@@ -320,7 +325,21 @@ def plot1d(
 
         import hist.plot
 
-        return hist.plot.histplot(self, ax=ax, **_proc_kw_for_lw(kwargs))
+        if self.ndim == 1:
+            return hist.plot.histplot(self, ax=ax, **_proc_kw_for_lw(kwargs))
+        else:
+            if overlay is not None:
+                (cat_ax,) = [ax for ax in self.axes if ax.name == overlay]
+            else:
+                (cat_ax,) = [ax for ax in self.axes if ax.traits.discrete]
```

Comment:
```suggestion
                (cat_ax,) = (ax for ax in self.axes if ax.traits.discrete)
```

No need to iterate twice.